language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -528,92 +570,74 @@ writePackageIdTs dir pkgId =
+      , "plugins" .= (["@typescript-eslint"] :: [T.Text])
+      , "extends" .= ([
+             "eslint:recommended"
+           , "plugin:@typescript-eslint/eslint-recommended"
+           , "plugin:@typescript-eslint/recommended"
+           , "plugin:@typescript-eslint/recommended-requiring-type-checking"
+           ] :: [T.Text])
+      , "rules" .= object [
+            ("@typescript-eslint/explicit-function-return-type", "off")
+          , ("@typescript-eslint/no-inferrable-types", "off") ]
+      ]
 
 writePackageJson :: FilePath -> SdkVersion -> Scope -> [Dependency] -> IO ()
 writePackageJson packageDir sdkVersion (Scope scope) depends =
-  writeFileUTF8 (packageDir </> "package.json") $ unlines
-  (["{"
-   , "  \"private\": true,"
-   , "  \"name\": \"" <> name <> "\","
-   , "  \"version\": \"" <> version <> "\","
-   , "  \"description\": \"Produced by daml2ts\","
-   , "  \"dependencies\": {"
-   , "    \"@daml/types\": \"" <> version <> "\","
-   ] ++ dependencies ++
-   [ "    \"@mojotech/json-type-validation\": \"^3.1.0\""
-    , "  },"
-    , "  \"scripts\": {"
-    , "    \"build\": \"tsc --build\","
-    , "    \"lint\": \"eslint --ext .ts src/ --max-warnings 0\""
-    , "  },"
-    , "  \"devDependencies\": {"
-    , "    \"@typescript-eslint/eslint-plugin\": \"^2.11.0\","
-    , "    \"@typescript-eslint/parser\": \"^2.11.0\","
-    , "    \"eslint\": \"^6.7.2\","
-    , "    \"typescript\": \"~3.7.3\""
-    , "  }"
-    , "}"
-    ])
+  BSL.writeFile (packageDir </> "package.json") $
+    encodePretty (packageJson (NpmPackageName name) (NpmPackageVersion version) dependencies)
   where
-    version = versionToString sdkVersion
+    version = versionToText sdkVersion :: T.Text
     name = packageNameOfPackageDir packageDir
-    dependencies = [ "    \"" <> pkg <> "\": \"" <> version <> "\","
-                   | d <- depends
-                   , let pkg = "@" ++ scope ++ "/" ++ T.unpack (undependency d)
-                   ]
-
-    -- From the path to a package like '/path/to/daml2ts/d14e08'
-    -- calculates '@daml2ts/d14e08' suitable for use as the "name" field
-    -- of a 'package.json'.
-    packageNameOfPackageDir :: FilePath -> String
+    dependencies = HMS.fromList [ (NpmPackageName pkg, NpmPackageVersion version)
+       | d <- depends
+       , let pkg = "@" <> scope <> "/" <> unDependency d
+       ]
+    packageNameOfPackageDir :: FilePath -> T.Text
     packageNameOfPackageDir packageDir = "@" <> scope <> "/" <> package
       where
-        scope = unscope $ scopeOfScopeDir (takeDirectory packageDir)
-        package = takeFileName packageDir
+        scope = unScope $ scopeOfScopeDir (takeDirectory packageDir)
+        package = T.pack $ takeFileName packageDir
+
+    packageJson :: NpmPackageName -> NpmPackageVersion -> HMS.HashMap NpmPackageName NpmPackageVersion -> Value
+    packageJson name version@(NpmPackageVersion sdkVersion) dependencies = object [
+        "private" .= True
```

Comment:
The idiomatic Haskell style with leading commas also puts the bracket at the beginning of the line.
```suggestion
    packageJson name version@(NpmPackageVersion sdkVersion) dependencies = object
      [ "private" .= True
```
If you want the bracket at the end of the previous lines, then it's more like
```haskell
foo = [
  bar,
  baz
]
```

Reply:
ok